shell/platform/common/accessibility_bridge.cc

```diff
@@ -111,6 +111,23 @@ AccessibilityBridge::GetPendingEvents() {
   return result;
 }
 
+void AccessibilityBridge::UpdateDelegate(
+    std::unique_ptr<AccessibilityBridgeDelegate> delegate) {
+  delegate_ = std::move(delegate);
+  // FlutterPlatformNodeDelegate[s] are created by the
+  // AccessibilityBridgeDelegate and may contain states from previous
+  // AccessibilityBridgeDelegate. Need to recreate them.
+  for (const auto& pair : id_wrapper_map_) {
+    std::shared_ptr<FlutterPlatformNodeDelegate> platform_node_delegate =
+        delegate_->CreateFlutterPlatformNodeDelegate();
+    platform_node_delegate->Init(
+        std::static_pointer_cast<FlutterPlatformNodeDelegate::OwnerBridge>(
+            shared_from_this()),
+        pair.second->GetAXNode());
+    id_wrapper_map_[pair.first] = platform_node_delegate;
```

Comment:
This should in term bring down the old FlutterPlatformNodeDelegate which will release the reference to old viewcontroller and restore any change made to the textinputplugin